clang/lib/3C/ProgramInfo.cpp

```diff
@@ -574,21 +591,23 @@ void ProgramInfo::addVariable(clang::DeclaratorDecl *D,
     NewCV = F;
     // Add mappings from the parameters PLoc to the constraint variables for
     // the parameters.
-    for (unsigned i = 0; i < FD->getNumParams(); i++) {
-      ParmVarDecl *PVD = FD->getParamDecl(i);
+    for (unsigned I = 0; I < FD->getNumParams(); I++) {
+      ParmVarDecl *PVD = FD->getParamDecl(I);
       const Type *Ty = PVD->getType().getTypePtr();
-      ConstraintVariable *PV = F->getParamVar(i);
-      unifyIfTypedef(Ty, *AstContext, PVD, dyn_cast<PVConstraint>(PV));
-      PV->setValidDecl();
+      PVConstraint *PVInternal = F->getInternalParam(I);
+      PVConstraint *PVExternal = F->getExternalParam(I);
+      unifyIfTypedef(Ty, *AstContext, PVD, PVInternal);
+      unifyIfTypedef(Ty, *AstContext, PVD, PVExternal);
+      PVInternal->setValidDecl();
```

Comment:
What does `setValidDecl` do? And why does it make sense to do it with the internal var?